tests/integration/test_lambda.py

```diff
@@ -755,8 +762,43 @@ def test_dotnet_lambda_running_in_docker(self):
         expected = ['Running .NET Core 2.0 Lambda']
         self.check_lambda_logs(TEST_LAMBDA_NAME_DOTNETCORE2, expected_lines=expected)
 
+
+class TestDotNetCore31Runtimes(LambdaTestBase):
+    @classmethod
+    def setUpClass(cls):
+        cls.lambda_client = aws_stack.connect_to_service('lambda')
+
+        # lambda .NET Core 3.1 is already a zip
+        zip_file = TEST_LAMBDA_DOTNETCORE31
+        cls.zip_file_content = None
+        with open(zip_file, 'rb') as file_obj:
+            cls.zip_file_content = file_obj.read()
+
+    @classmethod
+    def tearDownClass(cls):
         # clean up
-        testutil.delete_lambda_function(TEST_LAMBDA_NAME_DOTNETCORE2)
+        testutil.delete_lambda_function(TEST_LAMBDA_NAME_DOTNETCORE31)
+
+    def test_dotnet_lambda_running_in_docker(self):
+        if not use_docker():
+            return
+
+        testutil.create_lambda_function(
+            func_name=TEST_LAMBDA_NAME_DOTNETCORE31,
+            zip_file=self.zip_file_content,
+            handler='dotnetcore31::dotnetcore31.Function::FunctionHandler',
+            runtime=LAMBDA_RUNTIME_DOTNETCORE31
+        )
+        result = self.lambda_client.invoke(
+            FunctionName=TEST_LAMBDA_NAME_DOTNETCORE31, Payload=b'{}')
+        result_data = result['Payload'].read()
+
+        self.assertEqual(result['StatusCode'], 200)
+        self.assertEqual(to_str(result_data).strip(), '{}')
+
+        # assert that logs are present
+        expected = ['Running .NET Core 3.1 Lambda']
+        self.check_lambda_logs(TEST_LAMBDA_NAME_DOTNETCORE31, expected_lines=expected)
```

Comment:
Looks like this assertion is failing - we may have to make the `check_lambda_logs(..)` function a bit more resilient to different log formats:
```
======================================================================

FAIL: test_dotnet_lambda_running_in_docker (tests.integration.test_lambda.TestDotNetCore31Runtimes)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/localstack/localstack/tests/integration/test_lambda.py", line 801, in test_dotnet_lambda_running_in_docker

    self.check_lambda_logs(TEST_LAMBDA_NAME_DOTNETCORE31, expected_lines=expected)

  File "/home/travis/build/localstack/localstack/tests/integration/test_lambda.py", line 94, in check_lambda_logs

    self.assertIn(line, log_messages)

AssertionError: 'Running .NET Core 3.1 Lambda' not found in ['\x1b[32mSTART RequestId: 4426511a-e558-19e6-b969-b8464f027c18 Version: $LATEST\x1b[0m', 'Running .NET Core 3.1 Lambda\x1b[32mEND RequestId: 4426511a-e558-19e6-b969-b8464f027c18\x1b[0m', '\x1b[32mREPORT RequestId: 4426511a-e558-19e6-b969-b8464f027c18\tInit Duration: 138.22 ms\tDuration: 16.37 ms\tBilled Duration: 100 ms\tMemory Size: 1536 MB\tMax Memory Used: 39 MB\t\x1b[0m']
```

Reply:
I saw that and tried a few different things. is that something that I should do as part of this PR or comment it out and add a TODO (since the log itself is technically working).

Reply:
I saw that and tried a few different things. is that something that I should do as part of this PR or comment it out and add a TODO (since the log itself is technically working).

Reply:
Feel free to comment it out and add a TODO - we can pick it up in the next iteration 👍